### ISSUE TITLE
Firebase auth Username and Password prompt soft-lock-up

### ISSUE THREAD

**Issue by tfesslerCWC** [benwinding/react-admin-firebase#16]
With a browser, chrome or firefox, that is not authenticated to the react-admin-firebase project, initially navigating to the sites base-url of http://localhost:3000, the browser is redirected to http://localhost:3000/#/login.   When the username / Password screen is displayed, the Username and Password Fields as well as the sign in button are inactive,  there is a wait 'circle' animation over the sign in button that never ends.

there is an error in the javascript console:

Error: Missing or insufficient permissions. index.cjs.js:409
    FirestoreError index.cjs.js:409
    fromRpcStatus index.cjs.js:6971
    fromWatchChange index.cjs.js:7506
    onMessage index.cjs.js:17650
    startStream index.cjs.js:17561
    getCloseGuardedDispatcher index.cjs.js:17611
    newTail index.cjs.js:8990

Refreshing the browser window corrects the issue.

For reference here is the content of package.json
{
  "name": "material-test",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "react-admin": "2.9.1",
    "react-scripts": "3.0.1",
    "@material-ui/core": "3.9.3",
    "@material-ui/icons": "3.0.2",
    "css-loader": "2.1.1",
    "ra-data-fakerest": "2.6.0",
    "ra-input-rich-text": "2.8.6",
    "ra-language-english": "2.8.5",
    "ra-language-french": "2.8.5",
    "ra-core": "2.9.1"
  },
  "devDependencies": {
    "@babel/runtime": "7.0.0"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}

**Comment by benwinding** [comment]
Hi @tfesslerCWC,
Thanks for reporting this, but I'm unable to reproduce this on my machine (Firefox or Chrome) 
Can you try with `react-admin-firebase` version 0.5.6.
Please let me know how that goes, and if possible could you make a basic project on a github repository which demonstrates the issue.
Cheers,
Ben

**Comment by tfesslerCWC** [comment]
Yes, I can confirm that the issue still occurs with version 0.5.6 on MacOS in Chrome, Safari and Firefox.

<img width="757" alt="screenshot2019-05-227 23 16 AM" src="https://user-images.githubusercontent.com/7083751/58182622-cbd6c500-7c62-11e9-91b6-42650c7731d1.png">

**Comment by benwinding** [comment]
Thanks for responding again and reporting the issue. I'm unable to replicate this at this point, if you'd like to test the local demo in this repository and let me know if the problem still occurs on your system.

1. `git clone https://github.com/benwinding/react-admin-firebase`
2. `cd react-admin-firebase`
2. `yarn`
3. `yarn watch`
4. (open another terminal) `yarn start`

This should start the demo project, test it out and let me know if the same behaviour is happening.
Cheers,
Ben

**Comment by benwinding** [comment]
Closing this issue for now (might be fixed by #23), let me know if it occurs again.

**Comment by benwinding** [comment]
Unfortunately I can confirm that this issue is still present in version `0.6.5`. 

As @tfesslerCWC mentions, It appears to be a permissions problem, which in _Firestore's world_ means it could be authentication OR bad config OR bad document reference. 

**Reproducibility**

Not sure how to precisely reproduce the problem, but will update in the future.

**Tracing of problem for future reference**

![Selection_012](https://user-images.githubusercontent.com/11782590/59810736-ccd44280-9345-11e9-82bc-e1cbe5747cd0.png)

**Temporary fix**

Seems to be easily fixed by a page refresh.

**Comment by benwinding** [comment]
Update

### Reproducible?
Yes, here's how;

#### 1. Database rules

Must have the default rules on the Firestore database (which means you must log in):
``` javascript
service cloud.firestore {
  match /databases/{database}/documents {
    match /{document=**} {
      allow read, write: if request.auth.uid != null;
    }
  }
}
```

#### 2. Run app

1. Install `react-admin-firebase` version `0.6.5`: https://github.com/benwinding/react-admin-firebase/commit/fd6a0fc5d58903eab9c3edfb5c07cbf9825beb32
1. `yarn watch` ... new terminal `yarn start-demo`
1. Open a new browser **Incognito** window (from a normal browser window _not_ another incognito window)
1. Bug occurs!
![Selection_002](https://user-images.githubusercontent.com/11782590/59919277-2a5fb080-9466-11e9-9111-5f459f240b21.png)

It looks like it could be due to the logout function, but these are all executed asynchronously so it could be something else. This will take furthur investigation...

**Comment by benwinding** [comment]
This was a problem with the resource manager trying to initialize everything even if the user wasn't logged in. This should finally be fixed now in version `0.7.0`
